src/storage/RepresentationConvertingStore.ts

```diff
@@ -22,19 +22,23 @@ export class RepresentationConvertingStore<T extends ResourceStore = ResourceSto
   private readonly inPreferences: RepresentationPreferences;
 
   /**
-   * TODO: This should take RepresentationPreferences instead of a type string when supported by Components.js.
+   * @param source - Source store to get data from.
+   * @param metadataStrategy - Used to identify metadata resources as those always need to be converted.
+   * @param options - `outConverter` converts data returned by the store,
+   * `inConverter` converts incoming data,
+   * `inPreferences` determines how incoming data should be converted.
```

Comment:
I have a hard time remembering what this parameter means, and the text doesn't sufficiently help. Is there any clearer description possible?

Maybe "incoming" and "outcoming" are too vague. Do we mean data retrieved from the store, and data passes to the store?

```suggestion
   * @param source - Store we retrieve data from and send data to.
   * @param metadataStrategy - Used to distinguish regular resources (which may be converted) from metadata resources a (which always need conversion).
   * @param options.outConverter Converts data retrieved from the source store
   * @param options.inConverter Converts data passed to the source store
   * @param options.inPreferences The preferred input format for the source store, as passed to the inConverter
```

Still unsure why metadata always needs to be converted; isn't that specific to some configs?

Reply:
(It is interesting to see how time changes things 😄 https://github.com/CommunitySolidServer/CommunitySolidServer/pull/279#discussion_r514617339 )

Yes, incoming data = data from an HTTP request that we receive = data being sent to the resource store (e.g., if the backend is a SPARQL data accessor, we want to convert all incoming data into quads), and outcoming data = data we send in the response = data received from the store (so content negotiation).

> Still unsure why metadata always needs to be converted; isn't that specific to some configs?

Relevant comment (although I'm not sure how well this will load with how large that PR is): https://github.com/CommunitySolidServer/CommunitySolidServer/pull/1188#discussion_r915522844

The issue was that the `DataAccessorBasedStore` was taking a converter as input so it could convert the input for metadata resources to quads so it could store them in a `RepresentationMetadata`. It was then changed to doing it in this store instead since it specifically already had the job of converting incoming data for later stores.

Having looked at it again, I also have been thinking that this might not be flexible enough though. The alternative, more flexible, solution would be to create a new `ResourceStore` that converts incoming metadata to quads. This way that store can be removed again if someone has a `ResourceStore` solution where this step is not needed. Or to be even more flexible, what I'm currently thinking of:
 * `RepresentationConvertingStore` has an extra (optional) input `inValidator`, which is an `InputValidator` (better name pending).
 * `InputValidator` takes the identifier/representation as input and outputs a boolean.
 * `MetadataInputValidator` returns true if the target is a metadata identifier.
 * The `RepresentationConvertingStore` only applies the `inConverter` on inputs that return true on the validator.
 * Might even take a list of validators as input so you can have different preference/validator combinations.

But that would be better suited for a different PR though.

Reply:
> The alternative, more flexible, solution would be to create a new ResourceStore that converts incoming metadata to quads.

Made me wonder if metadata could be (conceptually) a different store, which is then a different wrapper around the internal store.

But yes, all out of scope for this PR.

Reply:
slightly modified from @RubenVerborgh's [suggestion](https://github.com/CommunitySolidServer/CommunitySolidServer/pull/1469/files#r1050196541)
```suggestion
   * @param source — Store we retrieve data from and send data to.
   * @param metadataStrategy — Used to distinguish regular resources (which may be converted) from metadata resources (which always need conversion).
   * @param options.outConverter — Converts data after retrieval from the source store
   * @param options.inConverter — Converts data before passing to the source store
   * @param options.inPreferences — The preferred input format for the source store, as passed to the inConverter
```